lib/benchee/formatters/json.ex

```diff
@@ -22,18 +22,17 @@ defmodule Benchee.Formatters.JSON do
 
   """
 
-  alias Benchee.{Suite, Statistics}
+  alias Benchee.{Suite, Configuration, Benchmark.Scenario}
 
   @doc """
   Uses `Benchee.Formatters.JSON.format/1` to transform the statistics output to
   a JSON, but also already writes it to a file defined in the initial
   configuration under `formatter_options: %{json: %{file: "my.json"}}`
   """
-  @spec output(Benchee.Suite.t) :: Benchee.Suite.t
-  def output(map)
-  def output(suite = %{configuration:
-                       %{formatter_options:
-                          %{json: %{file: filename}}}}) do
+  @spec output(Suite.t) :: Suite.t
+  def output(suite = %Suite{configuration: %Configuration{
+                        formatter_options: %{json: %{file: filename}}
+                      }}) do
```

Comment:
There's a Google Summer of Code project to write a code formatter for Elixir that José and Andrea are helping with! https://summerofcode.withgoogle.com/projects/#5205518009237504